Carousel: remove unused static images and tweak RTL styles

ramonjd opened the pull request:
Maybe resolves https://github.com/Automattic/jetpack/issues/20221 and resolves https://github.com/Automattic/view-design/issues/323

#### Changes proposed in this Pull Request:

Removing images directory, references to images and non-existent elements.

We're also tweaking arrows in RTL, and removing the `rtl/jetpack-carousel-rtl.css`, which seems to be referenced nowhere.

 Things currently look like this in RTL:

<img width="500" alt="Screen Shot 2021-07-08 at 4 35 45 pm" src="https://user-images.githubusercontent.com/6458278/124874015-8e804d80-e00a-11eb-833e-8bf52b85e239.png">

**Questions:**
- It looks like RTL styles are built automatically when we build. Is the stylesheet `projects/plugins/jetpack/modules/carousel/rtl/jetpack-carousel-rtl.css` redundant/have any effect at all?


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions

Check that no icons/images are missing.

In an RTL language, make sure things work and look as they should.

**LTR**
<img width="600" alt="Screen Shot 2021-07-08 at 4 28 21 pm" src="https://user-images.githubusercontent.com/6458278/124873741-347f8800-e00a-11eb-9f09-43861dc4d234.png">

**RTL**

<img width="600" alt="Screen Shot 2021-07-08 at 4 27 28 pm" src="https://user-images.githubusercontent.com/6458278/124873733-334e5b00-e00a-11eb-9824-acca72046dee.png">

ramonjd committed:
initial commit:
Removing images directory, references to images and non-existent elements

ramonjd committed:
Fixing RTL styles

ramonjd committed:
Changelog

glendaviesnz reviewed (dismissed):
Tested well for me - everything worked as expected on RTL and LTR language

jeherve commented on projects/plugins/jetpack/modules/carousel/jetpack-carousel.css:
```diff
@@ -96,10 +98,16 @@ so we have to target all affected elements in touch devices.
+
+.jp-carousel-overlay .swiper-container.swiper-container-rtl .swiper-button-prev,
+.jp-carousel-overlay .swiper-container.swiper-container-rtl .swiper-button-next {
+    transform: scaleX( -1 );
```
You'll want to avoid using spaces here, so no one changes it to a tab later when editing that file.

```suggestion
	transform: scaleX( -1 );
```

ramonjd replied:
Good spotting. 👍

jeherve reviewed (changes requested):
I cannot seem to get this to work. Here is what I see when using an RTL language like Hebrew:

![image](https://user-images.githubusercontent.com/426388/125252827-16ab6f00-e2f9-11eb-810d-0dc6c053b5e9.png)

ramonjd committed:
Update projects/plugins/jetpack/modules/carousel/jetpack-carousel.css

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

ramonjd commented:
> I cannot seem to get this to work. Here is what I see when using an RTL language like Hebrew:

Thanks for testing! 🙇 

When I was creating this PR, I noticed I had to rebuild the project at every change in order to regenerate  `jetpack/projects/plugins/jetpack/modules/carousel/jetpack-carousel-rtl.css`

Not sure why. 

As far as I can tell, the RTL styles are auto-generated via a [grunt-rtlcss](https://github.com/Automattic/jetpack/blob/9fcec83abd89719296e3cca40f9df31db27097ea/projects/plugins/jetpack/tools/builder/frontend-css.js#L26) job, and it's part of a [watch task](https://github.com/Automattic/jetpack/blob/79bcfcdb61cb572c8175f4b68056ead4bb91f12e/projects/plugins/jetpack/gulpfile.babel.js#L66). 🤷 

Here's what I'm seeing with a fresh, local build:

https://user-images.githubusercontent.com/6458278/125268302-ebea0680-e34a-11eb-9b0b-37e102cca746.mp4

This is probably a good opportunity to ask if there's any point in keeping `modules/carousel/rtl/jetpack-carousel-rtl.css` (the one in the `rtl` folder). It's outdated and doesn't seem to do anything. 

In the above example I deleted the `modules/carousel/rtl/` entirely before I rebuilt.

It appears we [replace](https://github.com/Automattic/jetpack/blob/95a7b30c34fcb57eadc352e03b4b21ab49af5446/projects/plugins/jetpack/modules/carousel/jetpack-carousel.php#L337) the CSS with the auto-generated `rtl.css` .

ramonjd committed:
Merge remote-tracking branch 'origin/master' into remove/carousel-image-directory

jeherve commented:
Hm yeah, there is something odd here.

Things seem to work when I declare `define( 'SCRIPT_DEBUG', true );`, because then I do not get to use the concatenated `jetpack-rtl.css`, but instead I get the `modules/carousel/jetpack-carousel-rtl.css` file.

I believe the concatenated `jetpack-rtl.css` will get rebuilt properly once #20274 is merged, so that should solve that problem.

That leaves us with `modules/carousel/rtl/`; like you said it may be time to delete it, since we changed things a few years back in #7407.

ramonjd committed:
Removing rtl/jetpack-carousel-rtl.css as it's not referenced anywhere

ramonjd commented:
> That leaves us with modules/carousel/rtl/; like you said it may be time to delete it, since we changed things a few years back in #7407.

Ah, good to know. Thanks. I've deleted it.

jeherve commented:
Cherry-picked to `jetpack/branch-9.9` in d1addec7e2f8adbc0416bc5c7b9a34bcf8140d31
